src/core.c

```diff
@@ -786,13 +786,24 @@ static int find_devices(Labels *data)
 		/* Looking for GPU */
 		if((data->gpu_count < LASTGRAPHICS / GPUFIELDS) && ((dev->device_class >> 8) == PCI_BASE_CLASS_DISPLAY))
 		{
+			const char *get_umd_base = "glxinfo -B | grep 'OpenGL core profile version'";
 			switch(dev->vendor_id)
 			{
-				case 0x1002: gpu_vendor = "AMD";    break;
-				case 0x8086: gpu_vendor = "Intel";  break;
-				case 0x10DE: gpu_vendor = "NVIDIA"; break;
-				default:     gpu_vendor = DEVICE_VENDOR_STR(dev);
-				             MSG_WARNING(_("Your GPU vendor is unknown: %s (0x%X)"), gpu_vendor, dev->vendor_id);
+				case 0x1002:
+					gpu_vendor = "AMD";
+					popen_to_str(&gpu_umd, "%s | sed -e 's/^.*Mesa/Mesa/g'",       get_umd_base);
+					break;
+				case 0x8086:
+					gpu_vendor = "Intel";
+					popen_to_str(&gpu_umd, "%s | sed -e 's/^.*Mesa/Mesa/g'",       get_umd_base);
+					break;
+				case 0x10DE:
+					gpu_vendor = "NVIDIA";
+					popen_to_str(&gpu_umd, "%s | sed -e 's/^.*NVIDIA/NVIDIA/g'",   get_umd_base);
+					break;
+				default:
+					gpu_vendor = DEVICE_VENDOR_STR(dev);
+					MSG_WARNING(_("Your GPU vendor is unknown: %s (0x%X)"), gpu_vendor, dev->vendor_id);
```

Comment:
You must not do that here: it assumes all AMD graphic card owners use Mesa, and all NVIDIA graphic card owners use proprietary driver.

I suggest you do it in `find_gpu_driver()` function (you can change the existing `switch(*gpu_driver)`.

Reply:
Thank you, I did it.  
https://github.com/X0rg/CPU-X/pull/183/commits/c179b2c30c7c8c5f2a847b7e7995bd62baa1c3b8